cmd/frontend/internal/app/debugproxies/scanner.go

```diff
@@ -198,10 +180,21 @@ func (cs *clusterScanner) scanCluster() {
 	cs.consume(scanResults)
 }
 
-// fromStrPtr returns *s. If s is nil the empty string is returned.
-func fromStrPtr(s *string) string {
-	if s == nil {
-		return ""
+// namespace returns the namespace the pod is currently running in
+// this is done because the k8s client we previously used set the namespace
+// when the client was created, the official k8s client does not
+func namespace() string {
+	const filename = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log15.Warn("unable to read ", filename, " using \"default\" ns")
```

Comment:
this isn't how log15 works, it is a structured logger.

```suggestion
		log15.Warn("endpoint: falling back to kubernetes default namespace", "filename", filename, "error", err)
```